exercises/bracket-push/example.scala

```diff
@@ -0,0 +1,23 @@
+import scala.util.parsing.combinator.RegexParsers
+
+object Brackets extends RegexParsers {
```

Comment:
Thanks for reviewing the solution.

It is probably a bit complex of a solution for the problem at hand. I wrote the example this way because I was thinking that just parsing for parens/brackets was not a real world problem. The real world problem would more likely be to perform some sort of expression or language parsing. The parens are just providing the scope. I figured that an example using parser combinators is more interesting than just showing how to parse using stacks or recursion.
Additionally I was trying not to just use recursion or iteration with a stack to solve the problem. 

I think I would rather nor provide a hint in the readme. I have no intent to force this solution on people. 

We could replace the solution with something more simple. Or, we could provide an alternate solution as another class. 

Here is a simple solution I wrote in Rust. I was trying not to just duplicate the iteration logic in Scala.
```rust

pub struct Brackets {
    s: String
}

impl Brackets {
    pub fn from(s: &str) -> Self {
        Brackets { s: s.to_string()}
    }

    pub fn are_balanced(self) -> bool {
        let mut stack = Vec::new();

        for c in self.s.chars() {
            match c {
                '(' | '[' | '{' => stack.push(c),
                ')' => if stack.is_empty() || stack.pop().unwrap() != '(' {
                            return false;
                        },
                ']' => if stack.is_empty() || stack.pop().unwrap() != '[' {
                            return false;
                        },
                '}' => if stack.is_empty() || stack.pop().unwrap() != '{' {
                            return false;
                        },
                _ => continue,
            }
        }

        return stack.is_empty();
    }
}```